airflow/models/deadline.py

```diff
@@ -90,3 +94,86 @@ def _determine_resource() -> tuple[str, str]:
     def add_deadline(cls, deadline: Deadline, session: Session = NEW_SESSION):
         """Add the provided deadline to the table."""
         session.add(deadline)
+
+
+class DeadlineTrigger:
+    """
+    Store the calculation methods for the various SDeadline Alert triggers.
+
+    TODO:  Embetter this docstring muchly.
+
+    usage:
+
+    In the DAG define a deadline as
+
+    deadline=DeadlineAlert(
+        trigger=DeadlineTrigger.DAGRUN_EXECUTION_DATE,
+        interval=timedelta(hours=1),
+        callback=hello,
+    )
+
+    to parse the deadline trigger use DeadlineTrigger.evaluate(dag.deadline.trigger)
+    """
+
+    DAGRUN_EXECUTION_DATE = "dagrun_execution_date"
+
+    @staticmethod
+    def evaluate(trigger: str):
+        return eval(f"DeadlineTrigger().{trigger}()")
+
+    @staticmethod
+    def get_from_db(table_name, column_name):
+        # TODO:
+        #   fetch appropriate timestamp from db
+        #   cast to datetime
+        #   return
+        log.info("MOCKED Getting %s :: %s", table_name, column_name)
+        return datetime(2024, 1, 1)
+
+    def dagrun_execution_date(self) -> datetime:
+        return self.get_from_db("dagrun", "execution_date")
+
+
+class DeadlineAlert(LoggingMixin):
+    """Store Deadline values needed to calculate the need-by timestamp and the callback information."""
+
+    def __init__(
+        self,
+        trigger: type[DeadlineTrigger] | datetime,
+        interval: timedelta,
+        callback: Callable | str,
+        callback_kwargs: dict | None = None,
+    ):
+        super().__init__()
+        self.trigger = trigger
+        self.interval = interval
+        self.callback_kwargs = callback_kwargs
+        self.callback = self.get_callback_path(callback)
+
+    @staticmethod
+    def get_callback_path(_callback: str | Callable) -> str:
+        if callable(_callback):
+            # Get the reference path to the callable in the form `airflow.models.deadline.get_from_db`
+            return f"{_callback.__module__}.{_callback.__qualname__}"
```

Comment:
This doesn't qite work right.    It works fine in the unit tests, but in a live DAG it returns 

```
"unusual_prefix_7308eaa9b565c9876e6fb24cd1ead1a9381ace55_dag_deadlines_test.missed_deadline"
```